lib/instrumentation/modules/@elastic/elasticsearch.js

```diff
@@ -0,0 +1,140 @@
+'use strict'
+
+// Instrument the @elastic/elasticsearch module.
+//
+// This uses to 'request' and 'response' events from the Client (documented at
+// https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/observability.html)
+// to hook into all ES server interactions.
+
+const { getDBDestination } = require('../../context')
+
+// TODO: ask Tomas about apiName addition to RequestEvent
+// Allow the following URL path components:
+// - _search
+// - _msearch
+// - _search/template
+// - _msearch/template
+// - _count
+// - _async_search
+const pathIsAQuery = /\/(_search|_msearch|_count|_async_search)(\/|$)/
```

Comment:
The `POST /_sql` path can contain a query as well. ([documentation](https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/api-reference.html#_sql_query))
```suggestion
const pathIsAQuery = /\/(_search|_msearch|_count|_async_search|_sql)(\/|$)/
```